tests/test_job.py

```diff
@@ -238,6 +238,12 @@ def test_deepcopy(self):
         copied_job.sp.a = 3
         assert copied_job in self.project
 
+    def test_project(self):
+        job = self.project.open_job({"a": 0})
```

Comment:
```suggestion
        job = self.project.open_job({"a": 0}).init()
```
We need to initialize job else it won't be registered into the workspace directory. This will also fix the failing tests.